frontend/main.py

```diff
@@ -44,9 +53,68 @@ def aws_cognito_redirect():
     response.set_cookie('access_token', access_token, expires=expires, httponly = True)
     return response
 
+@app.route('/corrections/<int:correction_id>', methods=['POST'])
+# @aws_auth.authentication_required
+def resolve_correction(correction_id):
+    req_data = request.form
+    if 'action' not in req_data:
+        return jsonify(message_maker("Please provide valid action param 'delete' or 'approve'"))
+    
+    action = req_data['action']
+
+    def delete_correction(correction_id):
+        dynamodb = boto3.resource('dynamodb',
+                    endpoint_url=app.config['DYNAMODB_ENDPOINT_URL'],
+                    region_name=app.config['REGION'])
+        table = dynamodb.Table(app.config['DYNAMODB_TABLE'])
+        try:
+            response = table.delete_item(Key={'id': str(correction_id)})
+        except ClientError as e:
+            return jsonify(message_maker(e.response['Error']['Message']))
+        
+        return jsonify(message_maker("Success"))
+
+    if action == "delete":
+        return delete_correction(correction_id)
+    
+    elif action == "approve":
+        dynamodb = boto3.resource('dynamodb',
+            endpoint_url=app.config['DYNAMODB_ENDPOINT_URL'],
+            region_name=app.config['REGION'])
+        table = dynamodb.Table(app.config['DYNAMODB_TABLE'])
+        try:
+            response = table.get_item(Key={'id': str(correction_id)})
+
+            conn = pymysql.connect(**mysql_properties)
+            cursor = conn.cursor()
+            query = "UPDATE bukhari_english SET hadithText = %(hadith_text)s WHERE englishURN = %(urn)s;"
+            cursor.execute(query, { "hadith_text": response['Item']['val'], "urn": response['Item']['urn'] })
+            rows_affected = cursor.rowcount
+            conn.commit()
+            conn.close()
+            
```

Comment:
like deletions, let's keep the approvals in another table that is used for both deletions and approvals. We can call that table HadithCorrectionLog